packages/twenty-front/src/modules/activities/emails/hooks/useEmailThread.ts

```diff
@@ -1,22 +1,37 @@
-import { useRecoilState } from 'recoil';
+import { useEffect, useState } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
 
 import { useOpenEmailThreadRightDrawer } from '@/activities/emails/right-drawer/hooks/useOpenEmailThreadRightDrawer';
 import { viewableEmailThreadIdState } from '@/activities/emails/state/viewableEmailThreadIdState';
+import { useRightDrawer } from '@/ui/layout/right-drawer/hooks/useRightDrawer';
+import { isRightDrawerOpenState } from '@/ui/layout/right-drawer/states/isRightDrawerOpenState';
 
 export const useEmailThread = () => {
-  const [, setViewableEmailThreadId] = useRecoilState(
+  const [viewableEmailThreadId, setViewableEmailThreadId] = useRecoilState(
     viewableEmailThreadIdState,
   );
-
+  const { closeRightDrawer } = useRightDrawer();
   const openEmailThredRightDrawer = useOpenEmailThreadRightDrawer();
+  const isRightDrawerOpen = useRecoilValue(isRightDrawerOpenState);
+  const [isDrawerActive, setIsDrawerActive] = useState(false);
+
+  useEffect(() => {
+    if (isRightDrawerOpen) {
+      setIsDrawerActive(true);
+      return;
+    }
+    setIsDrawerActive(false);
+  }, [isRightDrawerOpen]);
 
   const openEmailThread = (threadId: string) => {
+    if (viewableEmailThreadId === threadId && isDrawerActive) {
```

Comment:
Why not using directly isRightDrawerOpen ?

Reply:
Yeah you are right. When we click on an emailThread element and exclude it from the event listener in `useListenClickOutside`, it makes the isRightClose state clearer. This way, we can easily use isRightDrawOpened. I didn't realize this at first. I changed the code as you suggested. Thanks for your guidance and helping me understand the situation better.
```suggestion
     const toggleEmailThread = useRecoilCallback(
    ({ snapshot }) =>
      async (threadId: string) => {
        const isRightDrawerOpen = snapshot
          .getLoadable(isRightDrawerOpenState)
          .getValue();

        if (isRightDrawerOpen && viewableEmailThreadId === threadId) {
          setViewableEmailThreadId(null);
          closeRightDrawer();
        } else {
          setViewableEmailThreadId(threadId);
          openEmailThredRightDrawer();
        }
      },
  );
```